creme/creme_core/tests/gui/test_history.py

```diff
@@ -493,6 +511,35 @@ def test_render_edition_fk01(self):
 
         hline = self.get_hline()
         self.assertEqual(history.TYPE_EDITION, hline.type)
+
+        # new value format '["Hayao Miyazaki", ["position_id", X], ["image_id", X]]'
+        self.assertHTMLEqual(
+            '<div class="history-line history-line-edition">'
+            ' <ul>'
+            '  <li>{mod1}</li>'
+            '  <li>{mod2}</li>'
+            ' </ul>'
+            '<div>'.format(
+                mod1=_('{field} set to {value}').format(
+                    field=f'<span class="field-change-field_name">{_("Position")}</span>',
+                    value=f'<span class="field-change-new_value">{escape(position.title)}</span>',
+                ),
+                mod2=_('{field} set to {value}').format(
+                    field=f'<span class="field-change-field_name">{_("Photograph")}</span>',
+                    value=f'<span class="field-change-new_value">'
+                          f'<a href="{img.get_absolute_url()}" target="_self">'
+                          f'&lt;b&gt;Grumpy&lt;/b&gt; Hayao'
+                          f'</a>'
+                          f'</span>',
+                ),
+            ),
+            self.render_line(hline, user),
+        )
+
+        # old value format '["Hayao Miyazaki", ["position", X], ["image", X]]'
+        current_value = hline.value
+        self.remove_id_in_fk_modif(hline)
+
```

Comment:
I'd really prefer you create a new test case (e.g. `test_render_creation__old_format`) where you create an instance of HistoryLine manually (you assign explicitly `value='["Hayao Miyazaki", ["position", ...], ["image", ...]]' `etc...). So:

- we avoid a huge test case (we got 2 smaller test cases).
- we avoid to work with the same instance of `HistoryLine` (with internal modifications, backups...), so the test won't be successful because the underlying code use a cached value (& so get a useless test like before).
- we avoid to work with internal attribute like `_modifications`.

(Do not forget to remove the function `remove_id_in_fk_modif` which becomes useless).

Reply:
ok I're right !

I added the new test case `test_render_edition_fk01__old_fk_format` where I create manually an `HistoryLine` with the old version of `value`, inspired by the original test `test_render_edition_fk01`